jerixmx/nd064_course_1
# Project 1: TechTrends

### What this PR does (required):
- Adds health check, metrics, and logging to the application
- Provides docker commands for creating containers/images
- Sets up CI with Github Actions
- Provides Kubernetes manifests and Helm charts for CD
- Sets up CD with ArgoCD

### Screenshots / Videos (required):
- Refer to the [project's screenshots folder](https://github.com/jerixmx/CloudAppArchitecture/tree/project-techtrends/project/screenshots)
- [Demo](https://user-images.githubusercontent.com/78225194/130314635-5ec59d28-7750-4deb-9086-820c06a673cd.mp4)

### Any information needed to test this feature (required):
- Access `/healthz`
- Access `/metrics`
- Check console logs for events stated in the above section
- Start docker service and run commands listed in the [`docker_commands`](https://github.com/jerixmx/CloudAppArchitecture/blob/project-techtrends/project/docker_commands) file
- Create NodePort service and expose port
- Check application status on ArgoCD post-commit

### Any issues with the current functionality (optional):
- Redundant date/time in request info logs
